test/skimNtuple2016_HHbtag.cpp

```diff
@@ -599,51 +603,25 @@ int main (int argc, char** argv)
   TauIDSFTool * Deep_antiMu_tight       = new TauIDSFTool("2016Legacy","DeepTau2017v2p1VSmu" ,"Tight");    // for DeepTauv2p1 vs mu Tight
 
   // ------------------------------
-
-  // reweighting file for HH non resonant
-  TH1F* hreweightHH   = 0;
-  TH2F* hreweightHH2D = 0;
-  // if (HHreweightFile)
-  if (HHrewType == kFromHisto)
-    {
-      cout << "** INFO: doing reweight for HH samples" << endl;
-      if (HHreweightFile->GetListOfKeys()->Contains("hratio") )
-	{  
-	  hreweightHH = (TH1F*) HHreweightFile->Get("hratio");
-	  cout << "** INFO: 1D reweight using hratio" << endl;
-	}
-      else if (HHreweightFile->GetListOfKeys()->Contains("hratio2D") )
-	{
-	  hreweightHH2D = (TH2F*) HHreweightFile->Get("hratio2D");            
-	  cout << "** INFO: 2D reweight using hratio2D" << endl;
-	}
-      else
-	{
-	  cout << "** ERROR: reweight histo not found in file provided, stopping execuction" << endl;
-	  return 1;
-	}
-    }
-
-  // ------------------------------
-  // reweight file in case of "dynamic" reweight
+  // reweight file according to NLO differential reweighting procedure
   // there is a unique input map, read it from the cfg file
   // HHReweight* hhreweighter = nullptr;
   HHReweight5D* hhreweighter = nullptr;
   TH2* hhreweighterInputMap = nullptr;
-  if (HHrewType == kDynamic)
+  if (HHrewType != kNone)
     {
       string inMapFile   = gConfigParser->readStringOption("HHReweight::inputFile");
       string inHistoName = gConfigParser->readStringOption("HHReweight::histoName");
-      string coeffFile    = gConfigParser->readStringOption("HHReweight::coeffFile");
+      string coeffFile   = gConfigParser->readStringOption("HHReweight::coeffFileNLO");
+      if (order == "lo") coeffFile    = gConfigParser->readStringOption("HHReweight::coeffFileLO");
       cout << "** INFO: reading histo named: " << inHistoName << " from file: " << inMapFile << endl;
       cout << "** INFO: HH reweight coefficient file is: " << coeffFile << endl;
-      TFile* fHHDynamicRew = new TFile(inMapFile.c_str());
-      hhreweighterInputMap = (TH2*) fHHDynamicRew->Get(inHistoName.c_str());
-      // hhreweighter = new HHReweight(coeffFile, hhreweighterInputMap);
-      hhreweighter = new HHReweight5D(coeffFile, hhreweighterInputMap);
+      TFile* fHHDiffRew = new TFile(inMapFile.c_str());
+      hhreweighterInputMap = (TH2*) fHHDiffRew->Get(inHistoName.c_str());
+      if (HHrewType == kDiffRew) hhreweighter = new HHReweight5D(coeffFile, hhreweighterInputMap, EFTbm, string("2016"), order, uncertantie, cms_fake);
+      else hhreweighter = new HHReweight5D(coeffFile, hhreweighterInputMap, order, uncertantie);
```

Comment:
This is something I did not understand: what is the differnce between these two initialisations?

Reply:
When I wrote it, for some reason, I thought it was needed but looking at it know I agree with you -> removed